adr/ADR-8.md

```diff
@@ -214,6 +215,44 @@ type KV interface {
 }
 ```
 
+## KV Management
+
+This is set of operations on the KV stores from the JetStream context.
+
+```go
+// KeyValueManager is used to manage KeyValue stores. It provides methods to
+// create, delete, and retrieve.
+type KeyValueManager interface {
```

Comment:
Since this is an interface, the question is why make the names so complicated:
- create
- update
- list
- names
- destroy (delete is close to key operation, a `destroy` name is scary and that is a good thing

Reply:
Yeah probably could have been simpler but now consistency is probably better now, new languages could simplify I suppose

Reply:
The reason is - the KV interface is implemented by `jetstream context`.

That means you use it like this:

```go
var jetstream := jetstream.New(nc)
jetstream.CreateKeyValue()
```

```go
jetstream.Create()
```
Would not work well here.

Though that is pretty langauge dependant. I used it to be consistent across the doc. 
It's a really good point to add a notice that it should not be echo'y, depending on how given langauge is structured.

In Rust for example we have  modules structure:
```rust
jetstream::stream::Config
jetstream::consumer::pull::Config
```
so even struct names are not echo'y (so not `StreamConfig`), but again, that would not make much sense in Go.